tests/matrix-tests.lisp

```diff
@@ -29,6 +29,90 @@
+             (loop :for i :below 1000
+                   :do
+                      (let* ((n (1+ (random 5)))
+                             (m (1+ (random 5)))
+                             (k (1+ (random 5)))
+                             (a (magicl:rand (list m k)))
+                             (b (magicl:rand (list k n)))
+                             (c (magicl:rand (list m n))))
+                        ;; Check that multiplication returns the correct result
+                        (is (magicl:=
+                             (mult a b)
+                             (magicl:mult a b)))
+
+                        ;; Check that transposing doesn't affect correctness
+                        (is (magicl:=
+                             (mult (magicl:transpose a) c)
+                             (magicl:mult a c :transa :t)))
+                        (is (magicl:=
+                             (mult b (magicl:transpose c))
+                             (magicl:mult b c :transb :t)))
+                        (is (magicl:=
+                             (mult (magicl:transpose b) (magicl:transpose a))
+                             (magicl:mult b a :transa :t :transb :t)))
+
+                        ;; Check that alpha correctly scales the matrices
+                        (is (magicl:=
+                             (mult (magicl:scale a 2) b)
+                             (magicl:mult a b :alpha (coerce 2 magicl::*default-tensor-type*)))))))))
+
+(deftest test-matrix-vector-multiplication ()
+  "Test multiplication for random pairs of matrix and vectors"
+  (labels ((mult (a x)
+             (assert (= (magicl:ncols a) (magicl:size x)))
+             (let* ((m (magicl:nrows a))
+                    (n (magicl:ncols a))
+                    (target (magicl:empty (list m))))
+               (loop :for i :below m
+                     :do (setf (magicl:tref target i)
+                               (loop :for k :below n
+                                     :sum (* (magicl:tref a i k)
+                                             (magicl:tref x k)))))
+               target)))
+    (loop :for magicl::*default-tensor-type* :in +magicl-float-types+
+          :do
+             (loop :for i :below 1000
+                   :do
+                      (let* ((n (1+ (random 5)))
+                             (m (1+ (random 5)))
+                             (a (magicl:rand (list m n)))
+                             (x (magicl:rand (list n)))
+                             (y (magicl:rand (list m))))
+
+                        ;; Check that multiplication returns the correct result
+                        (is (magicl:=
+                             (mult a x)
+                             (magicl:mult a x)))
+
+                        ;; Check that transposing doesn't affect correctness
+                        (is (magicl:=
+                             (mult (magicl:transpose a) y)
+                             (magicl:mult a y :transa :t)))
```

Comment:
What does `:transa :t` mean?

Reply:
This specifies that you want to transpose the `A` argument. It is mimicking the LAPACK function call args (as with `ALPHA` and `BETA`)